### PR TITLE
Docs/improve seo

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

* Changed file locations to match urls
* Browser friendly, descriptive urls
* Added keywords, image and description meta tags
* Removed slugs

## Links to any relevant issues

* Resolves  iotaledger/docs.iota.org#35

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)
- [x] Documentation Fix

## How the change has been tested

Docs site was built and served locally using node 12.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by nothingismagick — APPROVED** [review]
Thanks!

**Comment by lucas-tortora** [comment]
@iotaledger/tech-writers , could someone review this ?

File: documentation/docs/contribute.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+---
+description: Contribute to Stronghold by joining the IOTA Stronghold Initiative, contributing to the GitHub repository and documentation, or sharing your knowledge on Discord.
```
Comment (aleksei-korolev) [inline documentation/docs/contribute.md]: Contribute by contributing

File: documentation/docs/retrospective.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 ---
-slug: /retrospective
+description: This document will detail the development of the Stronghold Engine library for IOTA's Stronghold project, touching upon different revisions and the lessons that were learnt from them.
```
Comment (aleksei-korolev) [inline documentation/docs/retrospective.md]:
```suggestion
description: History behind the development of the Stronghold Engine library that reflects upon previous revisions and the lessons learnt.
```

File: documentation/docs/retrospective.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+image: /img/logo/Stronghold_icon.png
+keywords:
+- crate
```
Comment (aleksei-korolev) [inline documentation/docs/retrospective.md:7]:
```suggestion
- development
- history
- crate
```

File: documentation/docs/specs/engineering.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
-# Specification: ENGINEERING
-
-# Stronghold Engineering Specification
+---
+description: This document introduces the High-Level Specification of the Stronghold.
```
Comment (aleksei-korolev) [inline documentation/docs/specs/engineering.md]:
```suggestion
description: This document introduces the high-level specification of Stronghold.
```

File: documentation/docs/specs/overview.md [hunk 5]
```diff
@@ -1,7 +1,12 @@
 ---
-slug: /specifications/overview
+description: This section contains a description of the scope of the project, and working documents that detail the engineering and requirements specifications.
```
Comment (aleksei-korolev) [inline documentation/docs/specs/overview.md]:
```suggestion
description: This section describes the scope of the project and includes working documents on engineering and requirements specifications.
```

File: documentation/docs/structure/derive.md [hunk 6]
```diff
@@ -1,3 +1,13 @@
+---
+description: The Structure:Derive crate contains procedural macros for Stronghold.
```
Comment (aleksei-korolev) [inline documentation/docs/structure/derive.md]:
```suggestion
description: The Derive crate contains procedural macros for Stronghold.
```

File: documentation/docs/structure/engine/store.md [hunk 7]
```diff
@@ -1,7 +1,16 @@
+---
+description: Store contains a key:value cache for the Stronghold Engine.
```
Comment (aleksei-korolev) [inline documentation/docs/structure/engine/store.md]:
```suggestion
description: Store contains a key-value cache for the Stronghold Engine.
```

File: documentation/docs/structure/overview.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+---
+description: In this section we describe the individual crates and components of Stronghold
```
Comment (aleksei-korolev) [inline documentation/docs/structure/overview.md]:
```suggestion
description: In this section we describe individual crates and components of Stronghold.
```

File: documentation/docs/structure/utils.md [hunk 9]
```diff
@@ -1,3 +1,11 @@
+---
+description: The Structure:utils crate contains common utilities for the stronghold libraries.
```
Comment (aleksei-korolev) [inline documentation/docs/structure/utils.md]:
```suggestion
description: The Utils crate contains common utilities for the stronghold libraries.
```

**Review by aleksei-korolev — CHANGES_REQUESTED** [review]

**Review by aleksei-korolev — APPROVED** [review]

**Pull request merged**